gettsim/generic_functions.py

```diff
@@ -57,6 +58,6 @@ def piecewise_linear(
 
     idx = np.searchsorted(upper_thresholds, value, side=side)
     intcpt = intercepts_at_lower_thresholds[idx]
-    out = intcpt + (value - intcpt) * rates[idx]
+    out = intcpt + (value - np.append(0, upper_thresholds)[idx]) * rates[idx]
```

Comment:
Please avoid any use of np.append. It is slow and incomprehensible.

But this looks like we may want to introduce an explict 0 indeed if the domain is >= 0; and -inf if it is the real line. Then it is guaranteed that upper_thresholds[idx -1] is defined. We might have to special-case 0 in the former case, though.